samples/estimation/df-chemistry/chemistry.py

```diff
@@ -0,0 +1,482 @@
+                # Remove only eigenvalues/vectors corresponding to `indices`
+                two_body_result.eigenvalues[r] = \
+                    np.delete(arr=two_body_result.eigenvalues[r], obj=indices)
+                arr = np.reshape(two_body_result.eigenvectors[r],
+                                 (orbitals, orbitals))
+                arr = np.delete(arr=arr, obj=indices, axis=0)
+                (rows, cols) = np.shape(arr)
+                two_body_result.eigenvectors[r] = np.reshape(arr, rows*cols)
+
+        # Check that same number of terms are removed for norms,
+        # eigenvalues and eigenvectors
+        assert len(two_body_result.one_norms) == \
+               len(two_body_result.two_norms) \
+               and len(two_body_result.one_norms) == \
+               len(two_body_result.eigenvalues) \
+               and len(two_body_result.one_norms) == \
+               len(two_body_result.eigenvectors)
+
+
+# Convert 2D array into string representation
+def ndarray2d_to_string(arr):
+    str_arr = []
+    for elem in arr:
+        str_arr.append(np.array2string(elem, separator=','))
+    return f"[{','.join(str_arr)}]"
+
+
+# The script takes one required positional argument, URI of the FCIDUMP file
+parser = ArgumentParser(description='Double-factorized chemistry sample')
+# Use n2-10e-8o as the default sample.
+# Pass a different filename to get estimates for different compounds
+parser.add_argument('-f', '--fcidumpfile',
+                    default='https://aka.ms/fcidump/n2-10e-8o',
+                    help='Path to the FCIDUMP file describing the Hamiltonian')
+args = parser.parse_args()
+
+# ----- Read the fcidump file and get Q# resource estimates -----
+structure = FCIDumpFileContent.from_file(args.fcidumpfile)
+df = DoubleFactorization.process_fcidump(structure, 0.001)
+
+# Load Q# project
+qsharp.init(project_root=".")
+
+# Construct the Q# string for which we need to perform resource estimate
+str_one_body_eigenvalues = np.array2string(df.one_body_eigenvalues,
+                                           separator=',')
+
+str_one_body_eigenvectors = ndarray2d_to_string(df.one_body_eigenvectors)
+
+str_two_body_eigenvalues = ndarray2d_to_string(df.two_body_eigenvalues)
+
+str_two_body_eigenvectors = "[" + \
+    ','.join([ndarray2d_to_string(eigenvectors)
+              for eigenvectors in df.two_body_eigenvectors]) + "]"
+
+qsharp_string = \
+    "Microsoft.Quantum.Applications.Chemistry.DoubleFactorizedChemistry(" + \
+    "Microsoft.Quantum.Applications.Chemistry.DoubleFactorizedChemistryProblem(" + \
+    f"{df.num_orbitals}, {df.one_body_norm}, {df.two_body_norm}, " + \
+    f"{str_one_body_eigenvalues}, {str_one_body_eigenvectors}, " + \
+    f"[1.0, size = {df.rank}], {str_two_body_eigenvalues}, " + \
+    f"{str_two_body_eigenvectors})," + \
+    "Microsoft.Quantum.Applications.Chemistry.DoubleFactorizedChemistryParameters(0.001,))"
+
+# Get resource estimates
+res = qsharp.estimate(qsharp_string,
```

Comment:
We could pass multiple qubit parameters by default and print a summary table.

Also, we could add a parameter to load a configuration from a local JSON file as an alternative to the default.

Reply:
Great suggestions, this would make a lot of sense to do as part of the next iteration of the sample along with the changes @billti proposed in https://github.com/microsoft/qsharp/pull/1033#discussion_r1457911557. For now, I'd like to focus on publishing the existing version with the core functionality, so that the users can start exploring it